src/textual/css/_color.py

```diff
@@ -0,0 +1,77 @@
+"""
+Helper functions to work with Textual Colors.
+"""
+
+from __future__ import annotations
+
+from textual.color import Color
+from textual.css._help_text import color_property_help_text
+from textual.css.errors import DeclarationError, StyleValueError
+from textual.css.scalar import Scalar, Unit
+from textual.css.tokenizer import Token
+
+
+def parse_color_tokens(property_name: str, tokens: list[Token]) -> tuple[Color, bool]:
+    """
+    Parses the tokens of a color expression, such as "red", "#FFCC00", "rgb(1,255,50)", "lime 50%", "95% auto"...
+
+    Args:
+        property_name (str): The name of the CSS property. Only used to raise DeclarationErrors that include this name.
+        tokens (list[Token]): CSS tokens describing a color - i.e. an expression supported by ``Color.parse``,
+            optionally prefixed or suffixed with a percentage value token.
+
+    Returns:
+        tuple[Color, bool]: a Color instance, as well as a boolean telling us if it's a pseudo-color "auto" one.
+
+    Raises:
+        DeclarationError: if we come across an unexpected value
+    """
+    property_name = property_name.replace("-", "_")
+
+    alpha = 1.0
+    color = None
+    is_auto = False
+
+    for token in tokens:
+        if token.value == "auto":
```

Comment:
It's not enough to look at just the value. We also need to change "name" as well, which I think should be "token".  

Consider if we add quoted strings to the parser, we wouldn't want to interpret `"auto"` in the same way as `auto`